sphinx/quickstart.py

```diff
@@ -375,14 +361,22 @@ def generate(d, overwrite=True, silent=False):
     d['project_manpage'] = d['project_fn'].lower()
     d['now'] = time.asctime()
     d['project_underline'] = column_width(d['project']) * '='
-    extensions = (',\n' + indent).join(
-        repr('sphinx.ext.' + name)
-        for name in EXTENSIONS
-        if d.get('ext_' + name))
+
+    extensions = []
+    if d.get('extensions'):
+        for ext in d['extensions'].split(','):
+            if ext not in ALLOWED_EXTS:
+                raise ValidationError("%s is not an allowed extension" % ext)
+            extensions.append("'sphinx.ext.%s'" % ext)
+    if d.get('contrib_extensions'):
+        for ext in d['contrib_extensions'].split(','):
+            extensions.append("'sphinxcontrib.%s'" % ext)
```

Comment:
It seems this does not support the 3rd party extensions which does not start with "sphinxcontrib"
(For example, breathe, sphinx-git and repoze.sphinx.autointerface).

I believe "sphinxcontrib" is a custom of package naming. But it is not a rule.
So we should not restrict its name.